diff --git a/httpclient/src/main/java/org/apache/http/client/entity/LazyDecompressingInputStream.java b/httpclient/src/main/java/org/apache/http/client/entity/LazyDecompressingInputStream.java
index 26e4981..ef07caf 100644
--- a/httpclient/src/main/java/org/apache/http/client/entity/LazyDecompressingInputStream.java
+++ b/httpclient/src/main/java/org/apache/http/client/entity/LazyDecompressingInputStream.java
@@ -57,6 +57,12 @@ class LazyDecompressingInputStream extends InputStream {
     }
 
     @Override
+    public int read(final byte[] b, final int off, final int len) throws IOException {
+        initWrapper();
+        return wrapperStream.read(b, off, len);
+    }
+
+    @Override
     public int available() throws IOException {
         initWrapper();
         return wrapperStream.available();
